common-libs/internal-api-clients/src/services/GenericApiService.ts

```diff
@@ -37,7 +37,7 @@ type RequestOptionsForOperation<TApi extends BuiltApiType, TOperationId extends
   RequestOptions<TApi[TOperationId]['requestBody'], TApi[TOperationId]['queryParams'], TApi[TOperationId]['pathParams']>
 >
 
-type ConstructorOptions = { accessApiKey: string }
+export type GenericConstructorOptions = { accessApiKey: string; headers?: NodeJS.Dict<string | string[]> }
```

Comment:
But it's `additionalHeaders`, not just `headers`.

And the goal of `internal-api-clients` is to isolate everything that has to do with HTTP from its users, so probably better to expect api version here rather than just generic headers.
(And that way we'll also be able to check that api version is always passed)

Reply:
To my opinion header `X-SDK-Version` should come exactly from `core` as version comes exactly from it's `package.json` and only `core` knows about this header.
Since `@affinidi/internal-api-services` is publicly available which means that anyone can use it without `@affinidi/wallet-core-sdk` and shouldn't know anything about sdk version (header), isn't it?

Reply:
We can accept the (required) user-agent string then.
And core can set it to e.g. `AffinidiCoreSdk/6.0.0`.

Reply:
Not sure understand how user-agent can help here, but the right thing that we should somehow distinguish requests from legacy sdk and without sdk, so then make sense have some default value (for e.g. `X-SDK-Version: unknown`) and move defining this header to `@affinidi/internal-api-services`

Reply:
User-agent can help in such a way that both SDK and other `internal-api-users` can pass something reasonable there, while `X-SDK-Version`, as you have mentioned, is specific to SDK.